lib/src/flutterflow_api_client.dart

```diff
@@ -265,3 +279,95 @@ Future _runFix({
     stderr.write('Error running "dart fix": $e\n');
   }
 }
+
+Future firebaseDeploy({
+  required String token,
+  required String projectId,
+  bool appendRules = false,
+  String endpoint = kDefaultEndpoint,
+}) async {
+  final endpointUrl = Uri.parse(endpoint);
+  final body = jsonEncode({
+    'project': {'path': 'projects/$projectId'},
+    'append_rules': appendRules,
+  });
+  final result = await _callEndpoint(
+    client: http.Client(),
+    token: token,
+    url: Uri.https(
+        endpointUrl.host, '${endpointUrl.path}/exportFirebaseDeployCode'),
+    body: body,
+  );
+
+  // Download actual code
+  final projectZipBytes = base64Decode(result['firebase_zip']);
+  final firebaseProjectId = result['firebase_project_id'];
+  final tmpFolder =
+      Directory.systemTemp.createTempSync('${projectId}_$firebaseProjectId');
+  final projectFolder = ZipDecoder().decodeBytes(projectZipBytes);
+  extractArchiveToCurrentDirectory(projectFolder, tmpFolder.path);
+  final firebaseDir = '${tmpFolder.path}/firebase';
+
+  try {
+    // Install required modules for deployment.
+    await Process.run(
+      'npm',
+      ['install'],
+      workingDirectory: '$firebaseDir/functions',
+      runInShell: true,
+      stdoutEncoding: utf8,
+      stderrEncoding: utf8,
+    );
+    final directoriesResult = tmpFolder.listSync(recursive: true);
```

Comment:
Nit: shouldn't `recursive` be `false`?

Reply:
replaced this by just checking if the directory exists